### PR TITLE
Fix voting dialogue

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

- Condition the dialogue to show only if user nodes exist
- Add getUserNodes

### Changes

List of changes this PR includes

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/839

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/dashboard/src/portal/components/VotePopup.vue [hunk 1]
```diff
@@ -20,22 +20,32 @@
...
 export default class VotePopup extends Vue {
   openProposalDialog = false;
   proposals = 0;
   voteMsg = "You have a pending proposal to vote!";
-
-  private __destroied = false;
+  $api: any;
+  farms: any = [];
+  farmsIds: any = [];
+  nodes: any = [];
```
Comment (Mahmoud-Emad) [inline packages/dashboard/src/portal/components/VotePopup.vue:36]: can we use typing instead of any?

Comment (zaelgohary) [reply packages/dashboard/src/portal/components/VotePopup.vue]: edited.

File: packages/dashboard/src/portal/components/VotePopup.vue [hunk 2]
```diff
@@ -44,6 +55,13 @@ export default class VotePopup extends Vue {
     this.openProposalDialog = true;
   }
 
+  async getUserNodes(farmIDs: any[]): Promise<any> {
+    const url = `${config.gridproxyUrl}/nodes?ret_count=true&page=1&size=10&farm_ids=${farmIDs}`;
```
Comment (AhmedHanafy725) [inline packages/dashboard/src/portal/components/VotePopup.vue:59]: we need also to make sure these nodes are online
```suggestion
    const url = `${config.gridproxyUrl}/nodes?status=up&ret_count=true&page=1&size=10&farm_ids=${farmIDs}`;
```

File: packages/dashboard/src/portal/components/VotePopup.vue [hunk 3]
```diff
@@ -44,6 +55,13 @@ export default class VotePopup extends Vue {
     this.openProposalDialog = true;
   }
 
+  async getUserNodes(farmIDs: any[]): Promise<any> {
```
Comment (AhmedHanafy725) [inline packages/dashboard/src/portal/components/VotePopup.vue:58]: we don't send the twin id to this method so I think the method name is invalid. maybe we can call it `getFarmsNodes`

Comment (AhmedHanafy725) [inline packages/dashboard/src/portal/components/VotePopup.vue:58]: we already know the type of the farmIDs
```suggestion
  async getUserNodes(farmIDs: number[]): Promise<any> {
```

File: packages/dashboard/src/portal/components/VotePopup.vue [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<template>
+<template v-if="nodes">
```
Comment (AhmedHanafy725) [inline packages/dashboard/src/portal/components/VotePopup.vue:1]: not sure if the `v-if` works with list length or not, can you make sure?

File: packages/dashboard/src/portal/components/VotePopup.vue (hunk 2 above)
Comment (AhmedHanafy725) [inline packages/dashboard/src/portal/components/VotePopup.vue:59]: the farmIDs here in the query are a list while the gridproxy is waiting for the farmIDs just numbers separated by a comma.

File: packages/dashboard/src/portal/components/VotePopup.vue (hunk 3 above)
Comment (AhmedHanafy725) [inline packages/dashboard/src/portal/components/VotePopup.vue:58]: we already know the return type, so we shouldn't use `any`

File: packages/dashboard/src/portal/components/VotePopup.vue [hunk 5]
```diff
@@ -20,22 +20,33 @@
...
 @Component({
   name: "VotePopup",
 })
 export default class VotePopup extends Vue {
   openProposalDialog = false;
   proposals = 0;
   voteMsg = "You have a pending proposal to vote!";
-
-  private __destroied = false;
+  $api: any;
```
Comment (AhmedHanafy725) [inline packages/dashboard/src/portal/components/VotePopup.vue:34]: the api type here is `ApiPromise` you can import it

`import { ApiPromise } from "@polkadot/api"`

File: packages/dashboard/src/portal/components/VotePopup.vue (hunk 4 above)
Comment (zaelgohary) [reply packages/dashboard/src/portal/components/VotePopup.vue:1]: Checked & it works.

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**